third_party/nvidia/include/Dialect/NVWS/IR/NVWSOps.td

```diff
@@ -175,4 +175,38 @@ def NVWS_WarpGroupReturnOp : NVWS_Op<"warp_group.return", [
   let assemblyFormat = "attr-dict";
 }
 
+def NVWS_CreateTokenOp : NVWS_Op<"create_token"> {
+  let results = (outs TensorOf<[NVWS_TokenType]>:$result);
+
+  let arguments = (ins I32Attr:$num, NVWS_TokenLoadTypeAttr:$loadType);
+
+  let builders = [OpBuilder<(ins "uint32_t":$num, "triton::nvws::TokenLoadType":$loadType)>];
+
+  let assemblyFormat = "attr-dict `:` type($result)";
+}
+
+def NVWS_ProducerAcquireOp : NVWS_Op<"producer_acquire"> {
+  let arguments = (ins TensorOf<[NVWS_TokenType]>:$token, I32:$idx, I1:$phase);
+
+  let assemblyFormat = "$token `,` $idx `,` $phase attr-dict `:` type(operands)";
+}
+
+def NVWS_ProducerCommitOp : NVWS_Op<"producer_commit"> {
```

Comment:
How similar do you think these ops are to https://github.com/triton-lang/triton/blob/7ff8e01ea78cc523c96a3ae39fa6958c37188c2a/include/triton/Dialect/TritonNvidiaGPU/IR/TritonNvidiaGPUOps.td#L692?

These are being added on NVIDIA's branch. cc @mbrookhart

Reply:
Honestly, these look a lot like the aref enter/exit ops from Nvidia's internal branch/what I just put up here:

https://github.com/triton-lang/triton/pull/6728

Main difference appears to be the inclusion of the  enum attribute for the token load type, and I know that @3gx is interested in adding enums to the aref implemenation.

The aref also contains ownership of the memory , which doesn't seem to be the case here. 

I am happy to add these if we think they are necessary, but I'm guessing we'll want to merge the implementations in the next month or two.

Reply:
Just to be clear, I don't want to block the upstreaming on preemptively unifying things, but I just wanted to get a signal on how aligned these abstractions are. Great to hear we are converging on the design here!